Medium/Triangle_shuna.java

```diff
@@ -0,0 +1,62 @@
+import java.util.List;
+
+/**
+ * 120. Triangle
+ * Given a triangle, find the minimum path sum from top to bottom. Each step you may move to adjacent numbers on the row below.
+ * For example, given the following triangle
+ * [
+ *    [2],
+ *   [3,4],
+ *  [6,5,7],
+ * [4,1,8,3]
+ * ]
+ * The minimum path sum from top to bottom is 11 (i.e., 2 + 3 + 5 + 1 = 11).
+ * Note:
+ * Bonus point if you are able to do this using only O(n) extra space, where n is the total number of rows in the triangle.
+ * Tags: Array Dynamic Programming
+ * Analysis: DP
+ * @author chenshuna
+ */ 
+
+public class Triangle {
+    public static int minimumTotal(List<List<Integer>> triangle) {
+        int minSum = Integer.MAX_VALUE;
+        int[] sum = new int[triangle.size()];
+        sum[0] = triangle.get(0).get(0);
+        for(int i = 1; i < triangle.size(); i++){  // from top to bottom
+            List<Integer> line = triangle.get(i);
+            for(int j = i; j >= 0; j--){
+                if(j == 0){
+                    sum[j] += line.get(j);
+                }
+                else if(j == i){
+                    sum[j] = sum[j - 1] + line.get(j);
+                }
+                else{
+                    sum[j] = Math.min(sum[j], sum[j - 1]) + line.get(j);
+                }
+            }
+        }
+        for(int cnt : sum){
```

Comment:
In this for-loop, you need to have extra space for each row at line 27, plus you need another for-loop to find the minimum at line 40.

It could be much easier if you try from bottom to top, you can initialize the `sum` array with the last row elements, adding all the way up to the top, finally you will find the minimum at sum[0].